### ISSUE TITLE
Update C# Implementation of JPatch

### ISSUE THREAD

**Issue by Mgamerz** [ME3Tweaks/ME3TweaksModManager#153]
In September 2020, Joris Heirbaut released a new version of JojoDiff, which very lightly altered the diff file format. While old patches will work on the new application tool, new diffs will not work on the old code. The implementation changes are as follows:
```
Hi,
The only change is that the ESC-MOD sequence is now the default when a new operation sequence is needed 
(at the start of a file or after an EQL, DEL or BKT operation). It's a small efficiency gain of 2 bytes, but also simplifies the code.
Thank's for using jdiff !
Best regards,
Joris Heirbaut
```

The JPatch.cs implementation I wrote needs to be updated to account for these so I can leverage the updated patch files, as the new diffing algorithm seems to be more efficient.

**Comment by mirh** [comment]
I see you are [already](https://github.com/janjongboom/janpatch/issues/16) working on it..
But by any chance, what's its usage here?
Couldn't [vcdiff](https://github.com/SnowflakePowered/vcdiff), [bsdiff](https://github.com/jzebedee/deltaq) or [xdiff](http://homac.cakelab.org/misc/xdiff/) be better?

**Comment by Mgamerz** [comment]
Cause on ME3Tweaks there is about 4200 patches in jdiff format already. No point in changing things since none of the old ones would work, plus I do not want external dependencies for it

**Comment by Mgamerz** [comment]
Implemented in 5e597602596ea692923332d7b0e0c7f2e3663f78

**Comment by mirh** [comment]
[Mhh I see](https://me3tweaks.com/modmanager/documentation/patchdesc).
